Git-commit: b043ae637a83585b2a497c2eb7ee49446fc68e98
From: "Eric W. Biederman" <ebiederm@xmission.com>
Date: Tue, 8 Mar 2022 13:01:19 -0600
Subject: [PATCH] coredump: Remove the WARN_ON in dump_vma_snapshot
References: bsn#19
Patch-mainline: v5.10.110

commit 49c1866348f364478a0c4d3dd13fd08bb82d3a5b upstream.

The condition is impossible and to the best of my knowledge has never
triggered.

We are in deep trouble if that conditions happens and we walk past
the end of our allocated array.

So delete the WARN_ON and the code that makes it look like the kernel
can handle the case of walking past the end of it's vma_meta array.

Reviewed-by: Jann Horn <jannh@google.com>
Reviewed-by: Kees Cook <keescook@chromium.org>
Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 fs/coredump.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/fs/coredump.c b/fs/coredump.c
index b18a6724be7a..b651675efe43 100644
--- a/fs/coredump.c
+++ b/fs/coredump.c
@@ -1128,12 +1128,6 @@ static bool dump_vma_snapshot(struct coredump_params *cprm)
 
 	mmap_write_unlock(mm);
 
-	if (WARN_ON(i != cprm->vma_count)) {
-		kvfree(cprm->vma_meta);
-		return false;
-	}
-
-
 	for (i = 0; i < cprm->vma_count; i++) {
 		struct core_vma_metadata *m = cprm->vma_meta + i;
 
-- 
2.35.1

